### PR TITLE
Create basic tests and configure phpunit.xml

### PR DESCRIPTION
**Pull request by malukenho**
- Create basic class for test Helper\Resource class
- Create a simple file called phpunit.xml with configurations to unit
  tests with PHPunit

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit bootstrap="vendor/autoload.php" colors="true">
+	<testsuites>
+		<testsuite name="Helper">
+			<directory>test</directory>
+		</testsuite>
+	</testsuites>
+</phpunit>
```
Comment (cordoval) [inline phpunit.xml]: end of line use php-cs-fixer

File: test/src/Helper/ResourceTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            array(null, $dirBase.'/../resources'),
+            array('can be storaged', 'can be storaged'),
+            array('//;sas/as', '//;sas/as')
+        );
+    }
+
+    /**
+     * @test
+     */ 
+	public function class_can_be_instaciate()
```
Comment (cordoval) [inline test/src/Helper/ResourceTest.php]: indentation

Comment (cordoval) [inline test/src/Helper/ResourceTest.php]: typo -> instantiated

File: test/src/Helper/ResourceTest.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Helper;
+
+use ReflectionObject;
+use Symfony\Component\Finder\Finder;
+
+class ResourceTest extends \PHPUnit_Framework_TestCase
+{
+
+    public function directory_provider()
```
Comment (cordoval) [inline test/src/Helper/ResourceTest.php:9]: check if maintainer likes this snake_case version, i think is ok, as long as it is verbalized and consistent

File: test/src/Helper/ResourceTest.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * @test
+     */ 
+	public function class_can_be_instaciate()
+	{
+		if (class_exists($className = __NAMESPACE__.'\Resource')) {
+            $this->assertInstanceOf(
+                $className,
+                new Resource, 
+                sprintf('Class %s cannot be instanciate', $className)
```
Comment (cordoval) [inline test/src/Helper/ResourceTest.php]: same typo

File: test/src/Helper/ResourceTest.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        $resource = new Resource($path);
+        $reflection = new ReflectionObject($resource);
+
+        $property = $reflection->getProperty('resources_dir');
+        $property->setAccessible(true);
+        $value = $property->getValue($resource);
+
+        $this->assertEquals($expected, $value);
+    }
+
```
Comment (cordoval) [inline test/src/Helper/ResourceTest.php]: remove line break

File: test/src/Helper/ResourceTest.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+}
```
Comment (cordoval) [inline test/src/Helper/ResourceTest.php:49]: php-cs-fixer

File: phpunit.xml (hunk 1 above)
Comment (cordoval) [inline phpunit.xml]: also indentation is typically 4 spaces, which IDE are you using? i use phpstorm

Comment (malukenho) [inline phpunit.xml]: I'am using `VIM` editor @cordoval

**Comment by malukenho** [comment]
@cordoval thanks for the recomendations :+1:

File: phpunit.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (henrikbjorn) [inline phpunit.xml]: it should be called phpunit.xml.dist so that a developer can have a local phpunit.xml with their own customizations.

Comment (malukenho) [inline phpunit.xml]: @henrikbjorn thanks for the info. There is some agreement as to that?

Comment (henrikbjorn) [inline phpunit.xml]: phpunit will always fallback on phpunit.xml.dist if it cannot find phpunit.xml so its a common practice to provide the phpunit.xml.dist file :)

Comment (malukenho) [inline phpunit.xml]: @henrikbjorn thanks again for the explanation :+1:

File: test/src/Helper/ResourceTest.php (hunk 2 above)
Comment (erikaheidi) [inline test/src/Helper/ResourceTest.php]: There's another typo, "storaged" => "stored"

File: test/src/Helper/ResourceTest.php (hunk 3 above)
Comment (erikaheidi) [inline test/src/Helper/ResourceTest.php:9]: For me its ok as long as its consistent and we stick with the snake_case for all test classes

File: test/src/Helper/ResourceTest.php (hunk 5 above)
Comment (erikaheidi) [inline test/src/Helper/ResourceTest.php]: method name - I guess this should be store ? "class can store resource... "

class_can_**store** (...)

**Comment by malukenho** [comment]
@cordoval @henrikbjorn I've corrected the errors mentioned

**Comment by erikaheidi** [comment]
This PR cannot be merged, because it was apparently based on an old commit or something. The file "Helper/Resource", for instance - modified in this PR - doesn't exists anymore, it was changed 3 months ago to src/ResourceHelper. Afaik it should be rebased, but since rebasing is a pain in the ass, it's better if you update your repo and then create a new PR later.

Closing

**Pull request closed without merging**